src/components/Avatar/Avatar.tsx

```diff
@@ -2,14 +2,56 @@ import * as PropTypes from 'prop-types'
 import * as React from 'react'
 import { Image, Label, Icon } from '../../'
 
-import { customPropTypes, UIComponent } from '../../lib'
+import { customPropTypes, UIComponent, NotStrictProps } from '../../lib'
+import { ComponentVariablesInput, IComponentPartStylesInput } from '../../../types/theme'
+
+export interface IAvatarPropsStrict {
```

Comment:
just as a matter of thought - it seems that we would be able to better reflect semantics behind these prop interfaces if we would name them like:

```jsx
interface IAvatarProps = {    // initially was 'IAvatarPropsStrict'
   alt?: string,
   as?: string,
   ...
}
...
// and later
class Avatar extends UIComponent<Extendable<IAvatarProps>, any> {
```

This approach seems to be something that introduces simplier names, better expressiveness (in terms of the goals that we would like to achieve), as well as provides generic `Extendable` name to generic

```
type Extendable<T> = T & { [key: string]: any }
```

rather than using specific `NonStrictProps` for this, in fact, generic concept.

What do you think?

Reply:
I agree on renaming the strict props to IAvatarProps and not having the current IAvatarProps. For the Extendable interface, I was looking for a name for the generic more describing about what is it in the context, but you are right that this might be use in other cases and the name would be inappropriate. So, in conclusion I will rename the NonStrictProps to Extendable and will get rid of having two interfaces for the props in the components. Will address in this PR the changes for the Accordion component as well.